homeassistant/components/renson/switch.py

```diff
@@ -0,0 +1,91 @@
+"""Breeze switch of the Renson ventilation unit."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from renson_endura_delta.field_enum import CURRENT_LEVEL_FIELD, DataType
+from renson_endura_delta.renson import Level, RensonVentilation
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RensonCoordinator
+from .const import DOMAIN
+from .entity import RensonEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RensonBreezeSwitch(RensonEntity, SwitchEntity):
```

Comment:
Can you maybe explain what the difference is between ventilation and breeze?

Reply:
Breeze is a functionality of the device that pushes the outside air into the home
Ventilation then their will be heat exchangers between the air outside and inside to ventilate the house. Ventilation will not cooldown the house but filters the air inside the house.

Reply:
Ah right, how does Breeze work with the different speed levels? Since you also have the fan mode breeze and you have several levels.

Reply:
Fan cannot work at the same time as Breeze function. The fan can set level of fan speed directly. For Breeze it is needed to change the Breeze configuration to change the levels.

Reply:
So if you can set the unit to Breeze via the fan entity, why do we need a separate switch?

(just trying to understand the why behind this change)

Reply:
We need a way to tell the device to use the breeze instead of the ventilation. The only way that I can think of to use only the fan for both ventilation and breeze if it is possible to set the speed level with negative values. Then we can use the negative values to the breeze and positive functions to ventilation (but don't think this is posible). Therefore this switch to enable the breeze function and use the ventilation level to configure the breeze level instead of the venitilcation level.

Reply:
Or we can create two fan sensors, one for ventilation and one for breeze. Than can also work

Reply:
Right, so if I understand correctly, you want to use the ventilation levels of the fan to control the level of breeze/ventilation based on the value of the switch?

Did I understand correctly?

Are there more modes than breeze/ventilation?

Reply:
That's correct. No there are no more modes that I know of.

Reply:
I think I got the picture now.

The thing I am doubting about is that a switch is usually for toggling stuff on and off. We don't really use them for mode A - mode B. I think in general we use select entities for this (there probably are cases in the code where we do use the switch, but that might be before the select was introduced).

I think merging both parts into one fan isn't ideal.

Two fans might work. as long as they can update correctly. So if I select level 3 on breeze, the other fan turns off.

I do still doubt if that is a correct way. So don't see this a definitive answer on what you should do, but more as a food for thought.

Reply:
I implented the fan approche.